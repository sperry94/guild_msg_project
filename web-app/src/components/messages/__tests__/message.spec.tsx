import React from "react"
import renderer from "react-test-renderer"
import Message, { MessageProps } from "../message"

describe("Message", () => {
  const cases: MessageProps[] = [
    {
      fromCurrentUser: true,
      time: new Date(2001, 1, 28, 12, 2, 5),
      message: "My message",
    },
    {
      fromCurrentUser: false,
      time: new Date(2019, 8, 13, 3, 5, 12),
      message: "My message #2",
    },
  ]

  cases.forEach((c, index) => {
    it(`renders for case #${index + 1}`, () => {
      const tree = renderer.create(<Message {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
