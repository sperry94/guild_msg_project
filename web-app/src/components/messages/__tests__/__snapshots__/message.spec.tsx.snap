// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders for case #1 1`] = `
<div
  className="makeStyles-outerContent-1 makeStyles-outerFromCurrentUser-3"
>
  <span
    className="MuiTypography-root makeStyles-time-7 makeStyles-timeFromCurrentUser-8 MuiTypography-caption"
  >
    2/28/2001, 12:02:05 PM
  </span>
  <div
    className="MuiPaper-root makeStyles-content-2 makeStyles-fromCurrentUser-5 MuiPaper-elevation3 MuiPaper-rounded"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      My message
    </p>
  </div>
</div>
`;

exports[`Message renders for case #2 1`] = `
<div
  className="makeStyles-outerContent-1 makeStyles-outerFromOtherUser-4"
>
  <span
    className="MuiTypography-root makeStyles-time-7 makeStyles-timeFromOtherUser-9 MuiTypography-caption"
  >
    9/13/2019, 3:05:12 AM
  </span>
  <div
    className="MuiPaper-root makeStyles-content-2 makeStyles-fromOtherUser-6 MuiPaper-elevation3 MuiPaper-rounded"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      My message #2
    </p>
  </div>
</div>
`;
